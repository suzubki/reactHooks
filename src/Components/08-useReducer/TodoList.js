import React from 'react'
import PropTypes from 'prop-types'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({ todos, handleDelete, handleIsDone }) => {
  return (
    <>
      <ul className='list-group list-group-flush'>
        {
          todos.map((todo, i) => (
            <TodoListItem
              key={todo.id}
              todo={todo}
              i={i}
              handleDelete={handleDelete}
              handleIsDone={handleIsDone}
            />
          ))
        }
      </ul></>
  )
}


TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleIsDone: PropTypes.func.isRequired
}

export default TodoList