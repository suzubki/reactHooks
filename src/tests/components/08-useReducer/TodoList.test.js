import TodoList from "../../../Components/08-useReducer/TodoList";
import { shallow } from "enzyme";

import { demoTodos } from "../../fixtures/demoTodos";

describe("<TodoList />", () => {
    const handleDelete = jest.fn();
    const handleIsDone = jest.fn();
    const wrapper = shallow(
        <TodoList
            todos={demoTodos}
            handleDelete={handleDelete}
            handleIsDone={handleIsDone}
        />
    );

    test("should display", () => {
        expect(wrapper).toMatchSnapshot();
    });
    test("should have components TodoListItem", () => {
        expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    });

    test("should have props", () => {
        expect(wrapper.find("TodoListItem").at(0).props().handleDelete).toEqual(
            expect.any(Function)
        );
        expect(wrapper.find("TodoListItem").at(0).props().handleIsDone).toEqual(
            expect.any(Function)
        );
    });
});
