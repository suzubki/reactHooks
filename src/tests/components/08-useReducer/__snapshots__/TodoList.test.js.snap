// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should display 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleIsDone={[MockFunction]}
      i={0}
      key="1"
      todo={
        Object {
          "desc": "Aprender react",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleIsDone={[MockFunction]}
      i={1}
      key="2"
      todo={
        Object {
          "desc": "Aprender mongo",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
